Covscript/test/logic.spec.ts

```diff
@@ -0,0 +1,22 @@
+import { expectEx } from "./common";
+
+describe("Logic expressions", () => {
+  describe("Basic logic expressions", () => {
+    expectEx("1 > 3", { ">": [1, 3] });
+    expectEx("2 < 5", { "<": [2, 5] });
+    expectEx("1 >= 1", { ">=": [1, 1] });
+    expectEx("2 <= 3", { "<=": [2, 3] });
+    expectEx('"Hello" == "Sir"', { "==": ["Hello", "Sir"] });
+    expectEx('"Goodbye" != "Madam"', { "!=": ["Goodbye", "Madam"] });
+    expectEx("!true", { "!": true });
+  });
+
+  describe("Logic operators", () => {
+    expectEx("true and true", { and: [true, true] });
+    expectEx("true or false", { or: [true, false] });
+  });
+
+  describe("Lested logic operators", () => {
+    expectEx("1 and 2 or 3 and 4", { or: [{ and: [1, 2] }, { and: [3, 4] }] });
```

Comment:
We could think about implementing something like:

expectEx("1 and 2 and 3", { and: [ 1, 2, 3 ] });

So, that operators like "and" or "+" are compacted into one array, which makes it easier to read/debug the json logic.

Reply:
Ohlala. Tricky but done. 

The following case (and others) are also handled nicely:

```
expectEx('true AND true OR false AND false OR true AND false', {
  or: [{ and: [true, true] }, { and: [false, false] }, { and: [true, false] }],
})
```